sparksteps/__main__.py

```diff
@@ -244,6 +245,9 @@ def main():
         pricing_client = boto3.client('pricing', region_name=args_dict['aws_region'])
         args_dict = determine_prices(args_dict, ec2_client, pricing_client)
         cluster_config = cluster.emr_config(**args_dict)
+        with open(args_dict['configuration']) as json_file:  
+          config = json.load(json_file)
+        cluster_config["Configurations"] = config        
```

Comment:
Unfortunately, this overrides the `Configurations` key in the dictionary, which is also being populated by [`cluster.emr_config`](https://github.com/jwplayer/sparksteps/blob/fb5b383acaf82f26b30fa44e270ae060cee88ab8/sparksteps/cluster.py#L112) and the `--defaults` argument.

Please move this logic into that function, and make sure that the `--defaults` argument remains backward compatible. In case both `--configuration` and `--defaults` is supplied, I would propose attempting to merge the two options into a single configuration object, where settings from `--defaults` take precedence (in order to remain backwards compatible).